syntax = "proto3";
package delphai;
import "validate/validate.proto";
import "google/api/annotations.proto";

service News {
  rpc add_article(AddArticleRequest) returns(AddArticleResponse) {
    option(google.api.http) = {post : "/"};
  }
}

message AddArticleRequest {
  string url = 1;
}

message AddArticleResponse {
  string article_id = 1;
  string title = 2;
  string content = 3;
  string date = 4;
  string translated_content = 5;
  string translated_title = 6;
  string lang = 7;
  repeated string mentions = 8;
}